source/MRMesh/MRMeshExtrude.cpp

```diff
@@ -0,0 +1,108 @@
+#include "MRMeshExtrude.h"
+#include "MRMesh.h"
+#include "MRRegionBoundary.h"
+#include "MRTimer.h"
+
+namespace MR
+{
+
+void makeDegenerateBandAroundRegion( Mesh& mesh, const FaceBitSet& region, FaceBitSet* outNewFaces, FaceMap* old2newMap )
+{
+    MR_TIMER
+
+    auto& topology = mesh.topology;
+    if ( region.none() || ( topology.getValidFaces() - region ).none() )
+        return;
+
+    auto componentBoundary = findLeftBoundaryInsideMesh( topology, region );
+    auto meshBoundary = componentBoundary;
+
+    FaceHashMap src2tgtFaces;
+    WholeEdgeHashMap src2tgtEdges;
+    PartMapping mapping {
+        .src2tgtFaces = old2newMap ? &src2tgtFaces : nullptr,
+        .src2tgtEdges = &src2tgtEdges,
+    };
+    auto component = mesh.cloneRegion( region, false, mapping );
+    for ( auto& contour : componentBoundary )
+    {
+        for ( auto& be : contour )
+        {
+            be = be.even() ? src2tgtEdges[be] : src2tgtEdges[be.sym()].sym();
+            assert( component.topology.isLeftBdEdge( be ) );
+        }
+    }
+
+    topology.deleteFaces( region );
+    for ( const auto& contour : meshBoundary )
+        for ( [[maybe_unused]] const auto be : contour )
+            assert( topology.isBdEdge( be ) );
```

Comment:
if whole block is needed only for assert mb put it under 
#ifdef NDEBUG ?

Reply:
Added the conditional.